backend/src/user/user.controller.ts

```diff
@@ -0,0 +1,86 @@
+import {Controller, Get, Post, Body, UseGuards, UseInterceptors, Param, Res, Put, Delete, Patch} from '@nestjs/common';
+import { UserService } from './user.service';
+import { RolesGuard } from '../common/guards/roles.guard';
+import { Roles } from '../common/decorators/roles.decorator';
+import { LoggingInterceptor } from '../common/interceptors/logging.interceptor';
+import { TransformInterceptor } from '../common/interceptors/transform.interceptor';
+import { ParseIntPipe } from '../common/pipes/parse-int.pipe';
+import {User} from './user.entity';
+import {CreateUserDto} from '@tsmean/shared';
+import {FindManyOptions} from 'typeorm';
+import {EmailValidatorImpl} from '../validation/email/email-validator.component';
+import {DeepPartial} from 'typeorm/common/DeepPartial';
+import {apiPath} from '../api';
+
+@Controller(apiPath(1, 'users'))
+@UseGuards(RolesGuard)
+@UseInterceptors(LoggingInterceptor, TransformInterceptor)
+export class UserController {
+  constructor(
+    private readonly userService: UserService,
+    private readonly emailValidator: EmailValidatorImpl
+  ) {}
+
+  @Post()
+  // @Roles('admin')
+  async create(@Body() requestBody: CreateUserDto, @Res() res) {
+
+    this.userService.create(requestBody.user, requestBody.password)
+      .then(data => {
+        res.status(200).send({
+          message: 'Success',
+          status: res.status,
+          data: data
+        });
+      })
+      .catch(err => {
+        if (err.message === 'User already exists') {
+          res.statusMessage = err.message;
+          res.status(403).send();
+        } else {
+          res.statusMessage = err.message;
+          res.status(500).send(err.message);
+        }
+      });
+  }
+
+  @Get()
+  // TODO: Only user can get info on himself or maybe admin
+  async find(options?: FindManyOptions<User>): Promise<User[]> {
+    const defaultOptions = {
```

Comment:
The same as earlier - one decision and I will refactor in all places 😉

Reply:
same as earlier

Reply:
@bersling You haven't responded to my comment https://github.com/tsmean/tsmean/pull/3#discussion_r167552422

Reply:
sorry, thought you referenced something else. commented now.